src/main/java/org/broadinstitute/hellbender/tools/spark/sv/Kmer.java

```diff
@@ -0,0 +1,177 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import java.io.Serializable;
+
+/**
+ * An immutable Kmer.
+ * K must be between 1 and 63 (but it's silly to use this class for K < 33).
+ * Canonicalization is unimplemented for even K.
+ *
+ * Created by tsharpe on 12/10/15.
+ */
+public final class Kmer implements Comparable<Kmer>, Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    // these are treated as K-bit unsigned integers
+    private final long v1; // most significant K bits
+    private final long v2; // least significant K bits
+
+    /**
+     *  Makes an empty Kmer.  If you call toString on it, it'll look like poly-A.
+     */
+    public Kmer( final int kSize )
+    {
+        if ( kSize < 1 || kSize > 63 ) throw new IllegalArgumentException("K must be between 1 and 63.");
+        v1 = v2 = 0;
+    }
+
+    /**
+     * Add a new trailing base to a Kmer, discarding the leading base.
+     * E.g., if kkk.toString(5) is "ACTGA", then kkk.successor(1,5).toString(5) is "CTGAC".
+     * @param base must be 0, 1, 2, or 3, corresponding to A, C, G, or T.
+     */
+    public Kmer successor( final long base, final int kSize )
```

Comment:
BaseUtils.Base is unsuitable.  It declares N and D (why D and no other ambiguity codes?) which are not legal for me.  I've declared a local enum in Kmer (next check-in).
It seems to me that append would be a very misleading name, since it's really a rotate operation (chop a base off the front, move everybody down, and add a new one to the back).  Happy to try to find a new verb, but I don't think append is the right one.